### ISSUE TITLE
Update rampup_batch_size usage to step_batch_size_schedule (Megatron-LM #3779)

### ISSUE THREAD

**Issue by yaoyu-33** [NVIDIA-NeMo/Megatron-Bridge#3438]
## Summary

Megatron-LM PR [NVIDIA/Megatron-LM#3779](https://github.com/NVIDIA/Megatron-LM/pull/3779) replaces the linear rampup batch size scheduler (`--rampup-batch-size` / `rampup_batch_size`) with a more flexible step-wise schedule (`--step-batch-size-schedule` / `step_batch_size_schedule`) and removes `rampup_batch_size` from public APIs (e.g. `init_num_microbatches_calculator`, `TrainingConfig`).

Megatron-Bridge currently still references `rampup_batch_size` in several places, which will break once we pick up the new MCore version.

cc @deepakn94 @ko3n1g

## Affected code

`src/megatron/bridge/training/initialize.py` — passes `train_config.rampup_batch_size` to `init_num_microbatches_calculator`:

```python
init_num_microbatches_calculator(
    get_rank_safe(),
    train_config.rampup_batch_size,
    train_config.global_batch_size,
    ...
)
```

`src/megatron/bridge/training/setup_mimo.py` — same call pattern via `getattr(cfg.train, "rampup_batch_size", None)`.

`src/megatron/bridge/training/config.py` — `TrainingConfig.rampup_batch_size` field and the sample-based-training assertion.

`examples/decentralized_pg/pretrain_qwen3_with_decentralized_pg.py` — passes `rampup_batch_size=None` (the example Oliver flagged).

Tests and docs that reference the field:
- `tests/unit_tests/training/test_config.py` (`test_sample_based_with_rampup_batch_size_fails`)
- `tests/unit_tests/training/mimo/test_pretrain_mimo.py`
- `tests/unit_tests/training/mimo/test_mimo_checkpointing.py`
- `docs/training/training-loop-settings.md`
- `docs/megatron-lm-to-megatron-bridge.md`

## Proposed fix

1. Replace the `rampup_batch_size: Optional[list[int]]` field on `TrainingConfig` with `step_batch_size_schedule: Optional[str]` matching the new MCore arg.
2. Update `init_num_microbatches_calculator` call sites in `initialize.py` and `setup_mimo.py` to the new signature.
   - Note from Deepak on the PR: `init_num_microbatches_calculator` may keep `rampup_batch_size` for backwards compatibility, so we may not strictly need a signature change there — but we should still migrate the field name on our side.
3. Update the example, tests, and docs accordingly.
4. Provide a short migration note in `docs/megatron-lm-to-megatron-bridge.md` showing how to convert an old `rampup_batch_size=[start, increment, ramp_samples]` config to a `step_batch_size_schedule` string.

## Context

From Slack discussion with Deepak and Oliver: the field is barely used in MB today, so this is a low-risk cleanup, but we need to land it before the next MCore bump or training will fail to initialize.
